src/site/content/en/payments/handling-optional-payment-information/index.md

```diff
@@ -0,0 +1,465 @@
+        details = await payment_request_event.changeShippingOption(newOption);
+        // Provided the new payment details,
+        // send a message back to the frontend to update the UI
+        postMessage('UPDATE_REQUEST', details);
+        break;
+…
+```
+
+Use the object to update the UI on the frontend. Since this object is
+consistently used across `.changePaymentMethod()`, `.changeShippingAddress()`
+and `.changeShippingOption()`, we'll cover this in the [Reflect the updated
+payment details](#reflect-the-updated-payment-details) section of this article.
+
+## Reflect the updated payment details {: #reflect-the-updated-payment-details }
+
+Once the merchant finishes updating the payment details, the promises returned
+from `.changePaymentMethod()`, `.changeShippingAddress()` and
+`.changeShippingOption()` will resolve with a common
+[`PaymentRequestDetailsUpdate`](https://w3c.github.io/payment-handler/#the-paymentrequestdetailsupdate)
+object. The payment handler can use the result to reflect updated total price
+and shipping options to the UI.
+
+{% Aside %}
+ The `PaymentRequestDetailsUpdate` object is passed from the merchant and it
+ contains the exact information the merchant put in the `PaymentDetailsUpdate`
+ object via the Payment Request API. The most important thing to handle things
+ gracefully is to provide good developer documentation so that merchants will
+ give you the right information you expect to receive. Or you may provide a
+ JavaScript library to control the expectation.
+{% endAside %}
+
+Beware that the merchant may return errors. Errors can be caused for a few reasons:
+* The payment method is not acceptable.
+* The shipping address is outside of their supported regions.
+* The shipping address contains invalid information.
+* The shipping option is not selectable for the provided shipping address or
+  some other reason
+
+Use `.error`, `.shippingAddressErrors`, or `.paymentMethodErrors` to reflect the error status.
+
+* **`error`**: Human readable error string. This is the best string to display
+  to customers.
+* **`shippingAddressErrors`**:
+  [`AddressErrors`](https://www.w3.org/TR/payment-request/#dom-addresserrors)
+  object that contains in-detail error string per address property. This is
+  useful if you want to open a form that lets the customer edit their address
+  and you need to point them directly to the invalid fields.
+* **`paymentMethodErrors`**: Payment-method-specific error object. You can ask
+  merchants to provide a structured error like
+  [`BasicCardErrors`](https://www.w3.org/TR/payment-method-basic-card/#dom-basiccarderrors)
+  for basic-card, but we recommend keeping it as simple as a simple string.
+
+## Next steps
+
+In this article, we learned how to handle optional information on the service
+worker. The final step for building a web-based payment app is to learn how to
+build the frontend.
+* Handling payments on the payment frontend (coming soon)
```

Comment:
```suggestion

* Handling payments on the payment frontend (coming soon)
```
I can't remember if I said this earlier but I always add newlines in front of lists because some Markdown engines seem to get mess up (i.e. they don't realize that it's a paragraph followed by an unordered list, but rather they literally put the `*` characters into the paragraph… it's not a problem with our current engine but I would prefer to futureproof our Markdown code